### PR TITLE
feat: LT-1008 - HMRC Organisation

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: parties/enums.py [hunk 1]
```diff
@@ -12,6 +12,18 @@ class PartyType:
...
 
 
+class OrganisationType:
+    HMRC = 'hmrc'
+    COMMERCIAL = 'commercial'
+    INDIVIDUAL = 'individual'
+
+    choices = [
+        (HMRC, 'Hmrc'),
+        (COMMERCIAL, 'Commercial Organisation'),
```
Comment (markm5A8Z9F43913A) [inline parties/enums.py]: Capitalisation of HMRC and Commercial organisation?

**Comment by markm5A8Z9F43913A** [comment]
Apart from that one comment, I'm willing to give this an approval as it is.

File: parties/enums.py (hunk 1 above)
Comment (cstobart) [reply parties/enums.py]: I'll fully capitalise HMRC I think that meets the pattern

File: organisations/views/main.py [hunk 2]
```diff
@@ -35,7 +47,7 @@ def post(self, request):
         """
         with reversion.create_revision():
             data = JSONParser().parse(request)
-            if data.get('sub_type') and data['sub_type'] == 'individual':
+            if data.get('type') and data['type'] == 'individual':
```
Comment (andytaylorkainos) [inline organisations/views/main.py]: i know all you did was change the key from `sub_type` to `type`. but can you get rid of one of the conditions in this statement? Theyre both the same thing

File: organisations/views/main.py [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
-        view_serializer = OrganisationViewSerializer(organisations, many=True)
-        return JsonResponse(data={'organisations': view_serializer.data})
+        org_type = self.request.query_params.get('org_type', None)
+        name = self.request.query_params.get('name', '')
+
+        query = [Q(name__icontains=name)]
+
+        if org_type:
+            query.append(Q(type=org_type))
+        return Organisation.objects.filter(reduce(operator.and_, query)).order_by('name')
```
Comment (andytaylorkainos) [inline organisations/views/main.py]: Can you not filter using this approach:
 `filter(**args)` where `args = {'name__contains': name, 'type': org_type}`? 
just curious

**Comment by efahy** [comment]
When creating a HMRC query, I get a CSRF exception when I click continue, having found and selected an org.  I realise that's somewhat beyond the scope of the story but for demoing, etc I'd prefer if you just disable the button or something

**Comment by efahy** [comment]
I don't see an option to enter details of a new exporter, which is part of the story.  Have I missed something?

File: organisations/views/main.py (hunk 3 above)
Comment (cstobart) [reply organisations/views/main.py]: type being optional throws a spanner in that

File: organisations/views/main.py (hunk 2 above)
Comment (cstobart) [reply organisations/views/main.py]: 👍

**Pull request merged**